apps/dcos_net/src/dcos_net_mesos_listener.erl

```diff
@@ -102,40 +103,49 @@ from_state(Data) ->
 %%%===================================================================
 
 init([]) ->
-    self() ! init,
-    {ok, []}.
+    Interval = application:get_env(dcos_net, mesos_reconnect_timeout, 2000),
+    MaxInterval = application:get_env(
+        dcos_net, mesos_reconnect_max_timeout, 30000),
+    Backoff0 = backoff:init(Interval, MaxInterval, self(), init),
+    Backoff = backoff:type(Backoff0, jitter),
+    {ok, #state{backoff = Backoff}, {continue, init}}.
 
 handle_call(is_leader, _From, State) ->
     % There is no state if it's not connected
```

Comment:
@ichernetsky it seems like now dcos_net_mesos_listener always thinks it's the leader, even if it runs on non-leader master nodes

Reply:
Nice catch. A fix is coming.

Reply:
BTW, is an immediate DC/OS bump with the fix warranted?

Reply:
it only affects the mesos dns exporter, `dcos_dns_mesos_dns`. So I guess we can wait for the PR that fixes partial mesos state.